dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Intrinsics.java

```diff
@@ -1328,6 +1331,106 @@ private List<Event> inlineMemCpy(FunctionCall call) {
+        Label check2 = EventFactory.newLabel("__memcpy_s_check_2");
+        Label success = EventFactory.newLabel("__memcpy_s_success");
+        Label end = EventFactory.newLabel("__memcpy_s_end");
+
+        // dest == NULL or destsz > RSIZE_MAX ----> return error > 0
+        final Expression c1 = expressions.makeOr(destIsNull, invalidDestsz);
+        CondJump skipE1 = EventFactory.newJump(expressions.makeNot(c1), check2);
+        CondJump skipRest1 = EventFactory.newGoto(end);
+        Local retError1 = EventFactory.newLocal(resultRegister, expressions.makeOne((IntegerType)resultRegister.getType()));        
+        replacement.addAll(List.of(
+            check1,
+            skipE1,
+            retError1,
+            skipRest1));
+
+        // dest != NULL && destsz <= RSIZE_MAX && (src == NULL || count > destsz || overlap(src, dest)) 
+        // ----> return error > 0 and zero out [dest, dest+destsz)
+        // The first two are guaranteed by not matching c1
+        final Expression c2 = expressions.makeOr(srcIsNull, invalidCount);
+        final Expression c3 = expressions.makeOr(c2, overlap);
+        CondJump skipE2 = EventFactory.newJump(expressions.makeNot(c3), success);
+        CondJump skipRest2 = EventFactory.newGoto(end);
+        Local retError2 = EventFactory.newLocal(resultRegister, expressions.makeOne((IntegerType)resultRegister.getType()));        
+        replacement.addAll(List.of(
+            check2,
+            skipE2));
+        for (int i = 0; i < destsz; i++) {
+            final Expression offset = expressions.makeValue(i, types.getArchType());
+            final Expression destAddr = expressions.makeAdd(dest, offset);
+            final Expression value = expressions.makeZero(types.getArchType());
+            replacement.addAll(List.of(
+                EventFactory.newStore(destAddr, value)
+            ));
+        }
+        replacement.addAll(List.of(
+            retError2,
+            skipRest2));
+
+        // Else ----> return error = 0 and do the actual copy
+        Local retSuccess = EventFactory.newLocal(resultRegister, expressions.makeZero((IntegerType)resultRegister.getType()));
+        replacement.add(success);        
+        for (int i = 0; i < count; i++) {
+            final Expression offset = expressions.makeValue(i, types.getArchType());
+            final Expression srcAddr = expressions.makeAdd(src, offset);
+            final Expression destAddr = expressions.makeAdd(dest, offset);
+            // FIXME: We have no other choice but to load ptr-sized chunks for now
+            final Register reg = caller.getOrNewRegister("__memcpy_" + i, types.getArchType());
+
+            replacement.addAll(List.of(
+                    EventFactory.newLoad(reg, srcAddr),
+                    EventFactory.newStore(destAddr, reg)
+            ));
+        }
+        replacement.addAll(List.of(
+            retSuccess,
+            end));
```

Comment:
Maybe it is just me, but I like to put the last brackets on a separate line:
```Java
       replacement.addAll(List.of(
            retSuccess,
            end
       ));
```
It's more symmetrical and matches better with how the standard curly brackets are placed.